packages/server/src/sdk/app/oauth2/tests/utils.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { generator } from "@budibase/backend-core/tests"
+import { GenericContainer, StartedTestContainer, Wait } from "testcontainers"
+import sdk from "../../.."
+import TestConfiguration from "../../../../tests/utilities/TestConfiguration"
+import { generateToken } from "../utils"
+import path from "path"
+import { KEYCLOCK_IMAGE } from "../../../../integrations/tests/utils/images"
+
+const config = new TestConfiguration()
+
+const volumePath = path.resolve(__dirname, "docker-volume")
+
+jest.setTimeout(60000)
+
+describe("oauth2 utils", () => {
+  let container: StartedTestContainer
+
+  let keycloakUrl: string
+
+  beforeAll(async () => {
+    await config.init()
+
+    container = await new GenericContainer(KEYCLOCK_IMAGE)
+      .withName("keycloak_testcontainer")
+      .withReuse()
+      .withExposedPorts(8080)
+      .withBindMounts([
+        { source: volumePath, target: "/opt/keycloak/data/import/" },
+      ])
+      .withCommand(["start-dev", "--import-realm"])
+      .withWaitStrategy(Wait.forLogMessage("Listening on: http://0.0.0.0:8080"))
+      .withStartupTimeout(60000)
+      .start()
+
+    keycloakUrl = `http://${container.getHost()}:${container.getMappedPort(
+      8080
+    )}`
```

Comment:
You _probably_ don't want to rely on `testcontainers` to get you the right port. We've seen trouble with that in the past around ipv6/ipv4. See the comment here: https://github.com/Budibase/budibase/blob/0e676f018365af79d602b1acef43131656ec3344/packages/backend-core/tests/core/utilities/testContainerUtils.ts#L186-L193

I recommend using that `startContainer` function here so you don't have to go through the same pain I did with running containers across local and CI. 🫠  It'll handle getting the ipv4 port so you can reliably connect to `127.0.0.1` (don't use localhost, you don't know if it'll resolve to a v4 or v6 address). It also handles setting the name and labels.

Reply:
I implemented the changes, but this does not support killing containers and other... Not sure where the original issue came from, but we have some usages (like in packages/backend-core/src/redis/tests/redis.spec.ts) that has been running for ages without issues. Not sure if the issue was a one-off one